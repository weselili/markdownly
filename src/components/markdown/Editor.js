import React from 'react';
import PropTypes from 'prop-types';
import styles from './Editor.css';

function Editor({ markdown, handleChange }) {
  return (
    <textarea className={styles.Editor} value={markdown} onChange={handleChange} />
  );
}

Editor.propTypes = {
  markdown: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired
};

export default Editor;
