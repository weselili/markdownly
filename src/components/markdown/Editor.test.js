import React from 'react';
import { shallow } from 'enzyme';
import Editor from './Editor';

describe('Editor component', () => {
  it('renders Editor', () => {
    const wrapper = shallow(<Editor markdown={''} handleChange={() => ({})}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
  
