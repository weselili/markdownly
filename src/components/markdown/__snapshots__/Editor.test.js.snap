// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor component renders Editor 1`] = `
<textarea
  className="Editor"
  onChange={[Function]}
  value=""
/>
`;
